### ISSUE TITLE
Assigning a blueprint with parameters file fails

### ISSUE THREAD

**Issue by Jandev** [Azure/azure-cli-extensions#2806]
## Describe the bug

**Command Name**
`az blueprint assignment create
Extension Name: blueprint. Version: 0.2.1.`

**Errors:**
```
Unable to build a model: Unable to deserialize to object: type, AttributeError: 'str' object has no attribute 'get', DeserializationError: Unable to deserialize to object: type, AttributeError: 'str' object has no attribute 'get'
Traceback (most recent call last):
Temp\pip-unpacked-wheel-kmz4tfp1\msrest\serialization.py, ln 1416, in _deserialize
Temp\pip-unpacked-wheel-kmz4tfp1\msrest\serialization.py, ln 1181, in rest_key_case_insensitive_extractor
...
```

## To Reproduce:
I'm using the Azure CLI on Windows in a PowerShell Core terminal.

```bash
az blueprint assignment create `
        --name 'sample-blueprint' `
        --identity-type SystemAssigned `
        --location westeurope `
        --parameters 'sample-blueprint-definition.parameters.json'
```

The parameter file has the following details:
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "blueprintName": {
            "value": "sample-blueprint"
        }
    }
}
```

The actual deployed & published blueprint has the following definition
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "blueprintName": {
            "defaultValue": "sample-blueprint",
            "type": "String",
            "metadata": {
                "description": "The name of the blueprint definition."
            }
        }
    },
    "variables": {
        "mgScope": "[tenantResourceId('Microsoft.Management/managementGroups', 'MVP')]"
    },
    "resources": [
        {
            "type": "Microsoft.Blueprint/blueprints",
            "apiVersion": "2018-11-01-preview",
            "name": "[parameters('blueprintName')]",
            "properties": {
                "targetScope": "subscription",
                "description": "Blueprint with a policy assignment artifact.",
                "resourceGroups": {
                    "demoRg": {
                        "description": "Some resource group."
                    }
                },
                "parameters": {
                    "effect": {
                        "type": "string",
                        "metadata": {
                            "displayName": "What to do."
                        },
                        "defaultValue": [
                            "Audit"
                        ]
                    }
                }
            }
        },
        {
            "type": "Microsoft.Blueprint/blueprints/artifacts",
            "apiVersion": "2018-11-01-preview",
            "name": "[concat(parameters('blueprintName'), '/policyArtifact')]",
            "kind": "policyAssignment",
            "dependsOn": [
                "[parameters('blueprintName')]"
            ],
            "properties": {
                "displayName": "OnlyAllowHTTPS",
                "description": "Only allow HTTPS traffic",
                "policyDefinitionId": "[extensionResourceId(variables('mgScope'), 'Microsoft.Authorization/policyDefinitions', 'EnforceHttpsOnlyWeb')]",
                "resourceGroup": "demoRg",
                "parameters": {
                    "effect": {
                        "value": "Deny"
                    }
                }
            }
        }
    ]
}
```
### Do note
This is my first time trying to create a blueprint via ARM templates. While deploying this exact template via the Portal I see there are some things that aren't great. Still, I'd expect a more clear error message, or at least not some Python exception.

## Expected Behavior

It would have been nice the blueprint got assigned to the provided subscription.
I can confirm the policy exists on both Management Group & Subscription level (because of debugging the issue) and I've deployed the blueprint like so:
```bash
az deployment sub create `
        --name demoDeployment `
        --location westeurope `
        --template-file "sample-blueprint-definition.json" `
        --parameters "sample-blueprint-definition.parameters.json"
az blueprint publish `
        --blueprint-name 'sample-blueprint' `
        --version '1.1'
```

## Environment Summary
```
Windows-10-10.0.19041-SP0
Python 3.6.8
Installer: MSI

azure-cli 2.14.0 *

Extensions:
blueprint 0.2.1

```

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls have a look? thanks

**Comment by anitian-mkimball** [comment]
I had this same issue and it turned out I was using the wrong schema for the parameters file. The example that I finally followed was here (in an obvious place, but I still missed it): https://docs.microsoft.com/en-us/azure/governance/blueprints/create-blueprint-azurecli

My successfully parsed parameters file looks something like the example except I used string literals for the values instead of a parameters() call:

{
   "tagName": {
      "value": "[parameters('tagName')]"
   },
   "tagValue": {
      "value": "[parameters('tagValue')]"
   }
}

**Comment by fengzhou-msft** [comment]
@Jandev the parameters file is not right. Please refer to this doc: https://docs.microsoft.com/en-us/azure/governance/blueprints/create-blueprint-azurecli

To make it more clear, I modified it bit from the example in the doc.
The `--parameters` in `blueprint create` should look like this:
```
{
   "bpTagName": {
       "type": "string",
       "metadata": {
           "displayName": "The name of the tag to provide the policy assignment.",
           "description": null
       }
   },
   "bpTagValue": {
       "type": "string",
       "metadata": {
           "displayName": "The value of the tag to provide the policy assignment.",
           "description": null
       }
   }
}
```
The `--parameters` in `blueprint artifact policy create` should look like this:
```
{
   "tagName": {
      "value": "[parameters('bpTagName')]"
   },
   "tagValue": {
      "value": "[parameters('bpTagValue')]"
   }
}
```
The `--parameters` in `blueprint assignment create` should look like this:
```
{
   "bpTagName": {
       "value": "CostCenter"
   },
   "bpTagValue": {
       "value": "ContosoIT"
   }
}
```

Basically, you define a parameter in blueprint (`bpTagName`), then you define some parameter in an artifact (`tagName`) which reference the parameter in blueprint (`[parameters('bpTagName')]`). When you assign the blueprint, you set value for the blueprint parameter (`bpTagName`), then the value will be used by artifact parameter (`tagName`).

**Comment by fengzhou-msft** [comment]
Let us know if you need further help.
